### PR TITLE
 Add additional pod and labels to generated skv2-generated helm chart deployment

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/skv2/issues/236

### CODE REVIEW

**Comment by ashleywang1** [comment]
> solo-io/gloo-mesh#1450

With the latest change, yes. The helm value would be `[serviceName].extraAnnotations`

**Comment by ashleywang1** [comment]
Tested with https://github.com/solo-io/gloo-mesh/pull/1499 and the helm template rendering worked.

File: codegen/templates/chart/operator-deployment.yamltmpl [hunk 1]
```diff
@@ -12,6 +12,8 @@ Expressions evaluating SKv2 Config use [[ "[[" ]] and [[ "]]" ]]
   [[- $workloadKind = "DaemonSet" ]]
   [[- end -]]
 [[- $operatorVar := (lower_camel $operator.Name) -]]
+[[- $extraPodLabels := $operator.Deployment.Labels -]]
+[[- $extraPodAnnotations := $operator.Deployment.Annotations -]]
```
Comment (harveyxia) [inline codegen/templates/chart/operator-deployment.yamltmpl:16]: Should we put these labels on the deployment as well? I'm not sure about the conventions around Pod labels vs. Deployment labels, usually I see the same set of labels on both

@ilackarms any thoughts?

Comment (ashleywang1) [reply codegen/templates/chart/operator-deployment.yamltmpl:16]: I'll add a `extraPodLabels` and `extraDeploymentLabels` field (same for annotations). This allows us to be flexible with which labels/annotations go where.

File: codegen/model/chart.go [hunk 2]
```diff
@@ -51,11 +51,14 @@ type Operator struct {
 // values for Deployment template
 type Deployment struct {
 	// use a DaemonSet instead of a Deployment
-	UseDaemonSet bool                     `json:"-"`
-	Labels       map[string]string        `json:"extraLabels,omitempty"`
-	Annotations  map[string]string        `json:"extraAnnotations,omitempty"`
-	Image        Image                    `json:"image,omitempty"`
-	Resources    *v1.ResourceRequirements `json:"resources,omitempty"`
+	UseDaemonSet    bool                     `json:"-"`
+	PodLabels       map[string]string        `json:"extraPodLabels,omitempty"`
```
Comment (harveyxia) [inline codegen/model/chart.go:55]: Why name the yaml field differently? Can we instead keep the names of the struct fields consistent with those of the yaml fields?

File: codegen/templates/chart/operator-deployment.yamltmpl [hunk 3]
```diff
@@ -12,8 +12,10 @@ Expressions evaluating SKv2 Config use [[ "[[" ]] and [[ "]]" ]]
   [[- $workloadKind = "DaemonSet" ]]
   [[- end -]]
 [[- $operatorVar := (lower_camel $operator.Name) -]]
-[[- $extraPodLabels := $operator.Deployment.Labels -]]
-[[- $extraPodAnnotations := $operator.Deployment.Annotations -]]
+[[- $extraPodLabels := $operator.Deployment.PodLabels -]]
```
Comment (harveyxia) [inline codegen/templates/chart/operator-deployment.yamltmpl:15]: Same comment here, though more nitpicky, can we use the same name for the template variable as we do for the struct field?

**Review by harveyxia — APPROVED** [review]

**Pull request merged**